### PR TITLE
New inference routine to mirror streaming-lda 0.6.2

### PR DESCRIPTION
**Pull request by charleslparker**
Here are the updates for the new version of streaming-lda that will be in wintermute after https://github.com/bigmlcom/wintermute/pull/987 goes through.

### CODE REVIEW

File: bigml/topicmodel.py [hunk 1]
```diff
@@ -308,32 +306,58 @@ def sample_uniform(self, term, rng):
...
+        topic = 0
+
+        while self.temp[topic] < random_value and topic < self.ntopics:
+            topic += 1
 
-        return self.sample_topic(term, assignments, norm, rng)
+        return topic
 
-    def infer(self, doc, max_updates):
+    def infer(self, list_of_indicies):
```
Comment (mmerce) [inline bigml/topicmodel.py]: typo? list_of_indices

File: bigml/topicmodel.py [hunk 2]
```diff
@@ -308,32 +306,58 @@ def sample_uniform(self, term, rng):
...
         """Infer a topic distribution for a document using `max_updates`
            iterations of gibbs sampling
 
         """
+
+        doc = sorted(list_of_indicies)
```
Comment (mmerce) [inline bigml/topicmodel.py]: ditto

File: bigml/topicmodel.py [hunk 3]
```diff
@@ -308,32 +306,58 @@ def sample_uniform(self, term, rng):
...
+
+        while self.temp[topic] < random_value and topic < self.ntopics:
+            topic += 1
 
-        return self.sample_topic(term, assignments, norm, rng)
+        return topic
 
-    def infer(self, doc, max_updates):
+    def infer(self, list_of_indicies):
         """Infer a topic distribution for a document using `max_updates`
```
Comment (mmerce) [inline bigml/topicmodel.py]: is it really max_updates or updates?

**Comment by mmerce** [comment]
I understand we should also change the test results in test_05_compare_predictions for LDA

**Comment by charleslparker** [comment]
> I understand we should also change the test results in test_05_compare_predictions for LDA

Yes - I think we have to wait for a deploy of wintermute for that, right?

**Comment by mmerce** [comment]
Right, but as this is not to be merged in bigmlcom/next right now, we can change that too here as soon as we have the new wintermute version installed in any development machine.

**Comment by charleslparker** [comment]
> Right, but as this is not to be merged in bigmlcom/next right now, we can change that too here as soon as we have the new wintermute version installed in any development machine.

Okay, yes.  As long as you're not planning to push other things to this branch in the next day or two we can do it here.

**Comment by mmerce** [comment]
I want to rebase the LDA code on top of a PR I sent yesterday, but that can wait.

**Comment by mmerce** [comment]
About the typo... I thought the plural  was indices or indexes, not indicies, but then I'm not american or english or australian... so you'll know better.

**Comment by charleslparker** [comment]
> About the typo... I thought the plural was indices or indexes, not indicies, but then I'm not american or english or australian... so you'll know better.

You're right.  I've been spelling it wrong for years.  Damn!

**Pull request merged**